kafka-rest/src/test/java/io/confluent/kafkarest/unit/ConsumerManagerTest.java

```diff
@@ -364,6 +364,119 @@ public void testMultipleTopicSubscriptionsFail() throws Exception {
+    config = new KafkaRestConfig(props, new SystemTime());
+    mdObserver = EasyMock.createMock(MetadataObserver.class);
+    consumerFactory = EasyMock.createMock(ConsumerManager.ConsumerFactory.class);
+    consumerManager = new ConsumerManager(config, mdObserver, consumerFactory);
+    expectCreateNoData();
+    EasyMock.expect(mdObserver.topicExists(topicName)).andReturn(true);
+    EasyMock.replay(mdObserver, consumerFactory);
+    String cid = consumerManager.createConsumer(groupName,
+            new ConsumerInstanceConfig(EmbeddedFormat.BINARY));
+    Future f = readTopicFuture(cid, topicName, Long.MAX_VALUE, new ConsumerReadCallback<byte[], byte[]>() {
+      @Override
+      public void onCompletion(List<? extends ConsumerRecord<byte[], byte[]>> records,
+                               RestException e) {
+        sawCallback = true;
+        actualRecords = records;
+        actualException = e;
+      }
+    });
+
+    // backoff is 250
+    Thread.sleep(100);
+    // backoff should be in place right now. the read task should be delayed and re-ran until the max.bytes or timeout is hit
+    System.out.println(consumerManager.delayedReadTasks.size());
+    Thread.sleep(100);
+    assertEquals(1, consumerManager.delayedReadTasks.size());
+    f.get();
+  }
+
+  @Test
+  public void testBackoffMsUpdatesReadTaskExpiry() throws Exception {
+    Properties props = new Properties();
+    props.setProperty(KafkaRestConfig.CONSUMER_ITERATOR_BACKOFF_MS_CONFIG, "1000");
+    // This setting supports the testConsumerOverrides test. It is otherwise benign and should
+    // not affect other tests.
+    props.setProperty("exclude.internal.topics", "false");
+    config = new KafkaRestConfig(props, new SystemTime());
+    mdObserver = EasyMock.createMock(MetadataObserver.class);
+    consumerFactory = EasyMock.createMock(ConsumerManager.ConsumerFactory.class);
+    consumerManager = new ConsumerManager(config, mdObserver, consumerFactory);
+    expectCreateNoData();
+    EasyMock.expect(mdObserver.topicExists(topicName)).andReturn(true);
+    EasyMock.replay(mdObserver, consumerFactory);
+    String cid = consumerManager.createConsumer(groupName,
+            new ConsumerInstanceConfig(EmbeddedFormat.BINARY));
+    Future f = readTopicFuture(cid, topicName, Long.MAX_VALUE, new ConsumerReadCallback<byte[], byte[]>() {
+      @Override
+      public void onCompletion(List<? extends ConsumerRecord<byte[], byte[]>> records,
+                               RestException e) {
+        sawCallback = true;
+        actualRecords = records;
+        actualException = e;
+      }
+    });
+
+    Thread.sleep(100);
+    ConsumerManager.RunnableReadTask readTask = consumerManager.delayedReadTasks.peek();
+    if (readTask == null) {
+      fail("Could not get read task in time. It should not be null");
+    }
+    long delay = readTask.getDelay(TimeUnit.MILLISECONDS);
+    assertTrue(delay < 1000);
+    assertTrue(delay > 700);
```

Comment:
Where does this value come from? Are there any concerns about this test becoming flaky?

Reply:
It's possible but I believe it shouldn't be flaky. I'll make the value more clear - it comes from the backoff.ms